### DISCUSSION TITLE
How to ignore a folder when running "next build"

### DISCUSSION THREAD

**Discussion by andy-asi** [vercel/next.js#11113] [Help] [answered]
When we run "next build" in our production server, it takes +20mins for the process to finish.
The problem for this delay is that we have a folder in the public/ directory called "uploads" and it has about 30,000 folders and files. This is our project hierarchy:

- ... other folders
- public/
   - ... other folders
   - uploads/     #### << here
      - ... (huge number of files)
- pages/

is there a way to prevent the build command from going through each folder and file inside the uploads folder because they are only used for serving static pictures and assets.

maybe adding an argument inside "next.config.js" to exclude specific folders?

thank you for your time!

**Comment by andy-asi** [comment]
does anyone have a solution for this?

**Comment by andy-asi** [comment]
still waiting for a solution

**Comment by janus-reith** [comment]
While I aggree such a feature could help migitating this, I guess it would be better to solve the underlying issues for now.
Why do you run "next build" on the production server from the very same folder things are served from?
Also, how are you doing version history with that - Is that folder simply ignored from your version history and you pull in changes?
I would try to separate that out somehow.

One thing you could do is switch to docker, and preferabbly, build the image which already contains the output of `next build` somewhere outside of that production server. Then you could mount that public folder as a docker volume.

**Comment by jaysinghcp** [comment]
facing same issue

**Comment by jguffey** [comment]
In the past I've done something like host user uploads in a AWS S3 bucket, instead of allowing users to upload assets to my source code. I think this is a safer approach and decouples user-uploaded files from the production server. I've only done this in Python, (Using this [module](https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html)) but I'm sure there's a parallel module in Node.

Good luck!

**Comment by protometa** [comment]
I also need to be able to tell Next to ignore certain files in the build process.

**Comment by timneutkens** [comment, marked as answer]
This issue was fixed on next@canary, please try it out.

**Reply by iKettles** [reply]
@timneutkens What exactly was fixed? Is there now a way to exclude files/folders from being included in production builds?

**Reply by timneutkens** [reply]
There was a bug where we would traverse the whole public directory to check if files conflict with routes in pages, this was fixed.

**Reply by iKettles** [reply]
Great, thanks for clearing that up.

**Reply by khelkun** [reply]
@timneutkens
I know it's a bit weird but I was looking for this because I have a repository where my nextjs app can be built as static (`output: "export"`) or not. So when it's built as static I just want the compiler to ignore the files under `src/app/api`. I've still not found a way to do that.

**Reply by washingtonsoares** [reply]
@khelkun I'm also facing this issue for the same use case

**Comment by eric-burel** [comment]
We still have a similar issue with any code colocated with sources, eg `__test__`, storybook stories, jest tests etc. The bugfix answers the precise question but not the question from the title.

For reference I need that to be able to remove test dependencies from my build and have slimmer docker images.

**Reply by eric-burel** [reply]
- https://github.com/vercel/next.js/issues/11144 is a dup of this discussion
- https://github.com/vercel/next.js/issues/10377, https://github.com/vercel/next.js/issues/8454, https://github.com/vercel/next.js/issues/3728 are limited to "pages/" folder
- https://github.com/vercel/next.js/issues/2026 closer to my issue, but they put "__tests__" in pages, whereas I put it at the root of my app so it shouldn't happen
- https://github.com/vercel/next.js/issues/1914 seems to provide potential solution in the comments... but went stale

What is included in the build seems to depend a lot on `tsconfig.json` when you use TypeScript. So excluding failing files seems to work for me.

However it gets me back to https://github.com/vercel/next.js/issues/12076 (configuring tsconfig path). The fact we need to exclude files from tsconfig.json for the build to work raises issues with VS Code, and the only solution that comes to my mind is having instead a `tsconfig.next.json`, which is not yet supported.

**Reply by dsauna** [reply]
@eric-burel did you figure this issue? I have something similar. I'm migrating a CRA application to nextjs, and would like next to have it's own tsconfig to ignore the react files not yet migrated.

**Reply by eric-burel** [reply]
Hi, not yet and I haven't investigated much how it could be done in Next. Can you just ignore them in the `tsconfig`? Then VS code may not handle them properly but at least the build would work

**Reply by paulm17** [reply]
Has this issue been resolved?  I have 13k files inside a folder in public (not user uploads) and I would like this ignored.  Thanks

**Reply by sbue** [reply]
Would love an update here as well.

**Reply by OzzieOrca** [reply]
Seems like using [Custom Page Extensions](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions) is a good solution. See:
https://github.com/vercel/next.js/issues/8454#issuecomment-560432659
https://github.com/vercel/next.js/issues/8617#issuecomment-790030032
https://github.com/vercel/next.js/issues/3728#issuecomment-523789071

**Reply by eric-burel** [reply]
Yeah but that's definitely hackish

**Reply by rtrembecky** [reply]
hi, popping in to verify - the tsconfig path can be configured in next.config.js as of the docs: https://nextjs.org/docs/basic-features/typescript#existing-projects

is this basically a best practice now? to have `tsconfig.build.json` (or `tsconfig.next.json`) for next builds and a separate `tsconfig.json` for IDE? where in `tsconfig.json` we would include all `.ts|.tsx` files and in `tsconfig.build.json` we would exclude storybook stories and setup, jest tests and setup, cypress tests and setup, other setup files, etc.?

if I'm right here, I feel like a simple mention of the config option in docs is not enough, this would better be fully explained

**Reply by OzzieOrca** [reply]
I've been able to colocate pages and other files in the `pages` directory by just changing `pageExtensions` in `next.config.js` and renaming all my pages to use the updated file extension. We got the docs updated at https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory. I haven't had to touch any tsconfig. What are you running into?

**Reply by rtrembecky** [reply]
I think I'm talking about the different part of this problem. Collocating tests and stories with pages is one thing, I'm actually using pageExtensions now as well.
The other thing is to prevent next.js build from even looking at the test and story files - At first, I was scared they might be included in the build, but I guess that's not the case. However, the first step of the build by default is typecheck, so I think whether it could be noticeably optimized by excluding some files explicitly in `tsconfig.build.json`.

**Reply by OzzieOrca** [reply]
Oh, gotcha. I didn't even think about that and am probably blindly type checking everything... Are you sure it's running typechecks on those files? Guess that'd be easy to test by breaking some type. I'd be curious to hear what other people are doing. Your multiple `tsconfig.json` files sounds like a valid solution... Might not be pretty.

**Reply by rtrembecky** [reply]
Just checked it and verified - I put a broken ts file in the `cypress/` folder and ran next build two times:
- with `tsconfigPath: "tsconfig.build.json",` in `next.config.js` and having `"exclude": ["cypress"]` in that tsconfig - typecheck was successful ✅ 
- without `tsconfigPath` in `next.config.js` - typecheck failed on the broken file ❌

**Reply by rtrembecky** [reply]
So it seems if you're not interested in typechecking tests/stories/config files during `next build`, the `tsconfigPath` is the way to go.

**Reply by eric-burel** [reply]
Nice to see `tsconfigPath`, it's a long awaited feature! For typechecking everything I go with a `tsc` as a CI step, it's a better pattern than typechecking during build.

**Comment by ctjlewis** [comment]
My workaround for this was adding the redundant folder to `exclude` in tsconfig.json, or overriding the `include` field to include only the directories you want.

You can still have your IDE type-check the directories you *don't* want Next to check by adding a local tsconfig:

<pre>
.
├── nextDontCompile
│   ├── index.tsx
│   └── <strong>tsconfig.json</strong> (local)
├── src
│   ├── index.tsx
│   └── utils.ts
└── <strong>tsconfig.json</strong> (Next)
</pre>

This seems messy, but it is ultimately what you want here: Your Firebase functions or whatever are really a *separate* project that are *not* part of the root-level Next project, and so it makes sense to ignore them.

You can also extend the root-level config to prevent having to duplicate your settings:

```none
# ./nextDontCompile/tsconfig.json

{
  "extends": "../tsconfig.json",
  ...
}
```

**Comment by artooras** [comment]
I would like to second a need for an option to ignore a directory when running `next build`. I tried following the above recommendations of simply reorganizing the project, however, this sometimes causes other issues. For instance, my Next.js project uses a number of Firebase services - auth, database, functions - along with emulators. If I move the `functions` folder outside of the root directory, emulators don't work properly anymore. And yes, it's likely possible to reconfigure the emulators to work even with such a setup, but it's not a mainstream use case, it certainly isn't well (if at all) documented, and would cost me a day or more researching, experimenting, etc. It would just be nice to just have an option for next build to ignore some directories in root and thus allow developers more flexibility in organizing their projects.

**Reply by ctjlewis** [reply]
@artooras, have you added the folders you don't want built to the `exclude` field in tsconfig? Or replace the `include` field with only the folders you want included?

Next runs a TS type-check and then builds the project (`pages/`, `api/`, etc.). The type-check stage *will* respect tsconfig (aside from a few fields which are overridden).

**Reply by eric-burel** [reply]
You'll lose typing of excluded files in VS Code. Worse, VS Code might use its default configuration, instead of your custom tsconfig, leading to unexpected conflicts (path aliases not respected etc.).

**Reply by ctjlewis** [reply]
@eric-burel You could create a local tsconfig in that directory, right? So that VS Code knows to type-check them for you, but also they're not tied into the root-level configuration, so Next won't touch them. See my example from above:

<pre>
.
├── nextDontCompile
│   ├── index.tsx
│   └── <strong>tsconfig.json</strong> (local)
├── src
│   ├── index.tsx
│   └── utils.ts
└── <strong>tsconfig.json</strong> (Next)
</pre>

I've been working in this part of the codebase recently and it literally just runs the equivalent of `tsc --noEmit` in the root directory, so if you tell TypeScript you don't want to include those things in the root project, you can avoid type-checking them.

Does this sound right? It's clear that there needs to be a help article or something on this at the very least and probably logic changes, just thinking out loud so I can consider the best way to approach this.

**Reply by eric-burel** [reply]
Yeah I must I haven't digged folders outside of src much, as my issue is more with collocated files (tests, stories), so multiple tsconfig should do it. It's quite hard to debug, but 2 tsconfig might do it in your case. `tsc --listFiles` is the way to go to test this.

**Reply by ctjlewis** [reply]
Agree. I have made the argument to the Next team that `src/` structure should be the default: This would simplify everything greatly by relying on the `rootDir` field. Currently in draft at #27667.

I will spend a few hours trying to write docs or otherwise make it easier for users to understand how to approach this, since it's not very intuitive at all.

Also IIRC, the logic excludes `test/` by default, as one of the few TS overrides.

**Reply by artooras** [reply]
Thanks for the suggestion @ctjlewis , this did indeed solve my issue. In my case it was also OK to `exclude` in the root's `tsconfig` as I was excluding the functions folder (which has it's own tsconfig and compiles locally anyway) and folders like tests, emulator-data, cypress, etc.

**Reply by ctjlewis** [reply]
cc @timneutkens—will send in docs to make approaching this clearer, see also draft #27667.

**Comment by coltenkrauter** [comment]
How can I ignore a folder when running `next build`? I didn't see an answer above, did I miss something?

**Reply by Xananax** [reply]
In theory, this should work... But it doesn't:

1. add it to the `ignore` section of `tsconfig.json`: 
   ```json
   ...
     "exclude": [
       "node_modules",
       "myDir"
   ]
   ```
   potentially also add a new `tsconfig.json` with and `"extends": "../tsconfig.json"` in the ignored directory
2. Add the following to `next.config.js`:
```js
//@ts-check

const rulesToProcess = [/\.m?js/, /\.(js|cjs|mjs)$/].map(String);
const dirToIgnore = /myDir/;

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  pageExtensions: ["js", "jsx", "ts", "tsx", "md", "mdx"],
  ...,
  /**
   *
   * @param {import('webpack').Configuration} config
   * @param {import('next/dist/server/config-shared').WebpackConfigContext} context
   * @returns {import('webpack').Configuration}
   */
  webpack: (config) => {
    config.module.rules = config.module.rules.map((rule) => {
      if (rule !== "..." && rulesToProcess.indexOf(String(rule.test)) > -1) {
        rule.exclude = [dirToIgnore];
      }
      return rule;
    });
    return config;
  },
};

export default nextConfig;
```

In my case, I am using `next.config.mjs` and I was trying to suppress the warning (https://github.com/vercel/next.js/discussions/30870, https://github.com/vercel/next.js/issues/33693, https://github.com/vercel/next.js/issues/27650). Ignoring the directory was convenient for me as it only contains build files, but alas I wasn't able to make it work.
I worked around the issue by suppressing webpack warnings, as described in https://github.com/vercel/next.js/discussions/30870#discussioncomment-1862620, but I'd rather do it more cleanly.

Still, I leave this here in the hopes it puts someone on some kind of correct path.

**Comment by luixo** [comment]
Made a poor man's patch to implement the feature of ignoring pages on build:
```diff
diff --git a/dist/build/entries.js b/dist/build/entries.js
index a455fef61988ef553069946b1138ca8dec7e9918..12d4682124c8a089d7b98cfbe77b39a2e4a53c92 100644
--- a/dist/build/entries.js
+++ b/dist/build/entries.js
@@ -153,7 +153,7 @@ function getPageFilePath({ absolutePagePath , pagesDir , appDir , rootDir  }) {
     }
     return require.resolve(absolutePagePath);
 }
-function createPagesMapping({ isDev , pageExtensions , pagePaths , pagesType , pagesDir  }) {
+function createPagesMapping({ isDev , pageExtensions , ignorePages , pagePaths , pagesType , pagesDir  }) {
     const isAppRoute = pagesType === "app";
     const previousPages = {};
     const pages = pagePaths.reduce((result, pagePath)=>{
@@ -161,6 +161,10 @@ function createPagesMapping({ isDev , pageExtensions , pagePaths , pagesType , p
         if (pagePath.endsWith(".d.ts") && pageExtensions.includes("ts")) {
             return result;
         }
+        const ignorePagesArray = Array.isArray(ignorePages) ? ignorePages : ignorePages ? [ignorePages] : [];
+        if (ignorePagesArray.some((regex) => regex.test(pagePath))) {
+            return result;
+        }
         let pageKey = getPageFromPath(pagePath, pageExtensions);
         if (isAppRoute) {
             pageKey = pageKey.replace(/%5F/g, "_");
diff --git a/dist/build/index.js b/dist/build/index.js
index 667a9efe1a4c24ba087a5d7a5ac8e8e6fb15ca10..ad1d5fd36d7732b229a8fa67110ea936870736dd 100644
--- a/dist/build/index.js
+++ b/dist/build/index.js
@@ -275,6 +275,7 @@ async function build(dir, reactProductionProfiling = false, debugOutput = false,
             const mappedPages = nextBuildSpan.traceChild("create-pages-mapping").traceFn(()=>(0, _entries.createPagesMapping)({
                     isDev: false,
                     pageExtensions: config.pageExtensions,
+                    ignorePages: config.ignorePages,
                     pagesType: "pages",
                     pagePaths: pagesPaths,
                     pagesDir
diff --git a/dist/server/config.js b/dist/server/config.js
index e9232bcb5dc81256f80ca82076384e68d009301a..2aa9c55635fe0f424cd195b64c99615658a55ddd 100644
--- a/dist/server/config.js
+++ b/dist/server/config.js
@@ -172,6 +172,21 @@ function assignDefaults(dir, userConfig, silent = false) {
                 }
             });
         }
+        if (key === "ignorePages") {
+            if (Array.isArray(value)) {
+                if (!value.length) {
+                    throw new Error(`Specified ignorePages is an empty array. Please update it with the relevant RegExps or remove it.`);
+                }
+                value.forEach((regex)=>{
+                    if (!(regex instanceof RegExp)) {
+                        throw new Error(`Specified ignorePages is not an array of RegExps, found "${regex}" of type "${typeof regex}". Please update this config or remove it.`);
+                    }
+                });
+            } else if (!(value instanceof RegExp)) {
+                throw new Error(`Specified ignorePages is neither an array nor a RegExp, found "${value}". Please update this config or remove it.`);
+            }
+            
+        }
         if (!!value && value.constructor === Object) {
             currentConfig[key] = {
                 ..._configshared.defaultConfig[key],
```

Also, you'll have to dig into `dist/next-config-validate.js` which is compiled beforehand for some reason, find `pageExtensions` and insert before it something like `ignorePages:{anyOf: [{instanceof: "RegExp"},{minItems:1,type:"array"}]}` (didn't include the patch here as it's replacing the whole one-line file).

This patch doesn't work in dev environment, you will still be able to open an ignored page while developing.

The `ignorePages` option accept `RegExp` or an array of `RegExp`s.
I personally use it to ignore tests that are co-located with pages: `ignorePages: /\.spec\.tsx?$/`

**Reply by luixo** [reply]
An updated patch for Next.js 14.0.3 (also I switched to `string | string[]` in `ignorePages` type as it seems like maintainers prefer to keep it serializable).
Also, no need to lookup prebuilt `next-config-validate.js` anymore.

My config prop now looks like this: `ignorePages: "\\.spec\\.tsx?$"`.

```diff
# see https://github.com/vercel/next.js/discussions/11113#discussioncomment-7083801
diff --git a/dist/build/entries.js b/dist/build/entries.js
index c759abdd2419661dd658c0f6741eca30e8000042..ce25d5a6e4a098391ff86d019e4b58dd47f41448 100644
--- a/dist/build/entries.js
+++ b/dist/build/entries.js
@@ -177,7 +177,7 @@ function getPageFilePath({ absolutePagePath, pagesDir, appDir, rootDir }) {
     }
     return require.resolve(absolutePagePath);
 }
-function createPagesMapping({ isDev, pageExtensions, pagePaths, pagesType, pagesDir }) {
+function createPagesMapping({ isDev, pageExtensions, ignorePages, pagePaths, pagesType, pagesDir }) {
     const isAppRoute = pagesType === "app";
     const previousPages = {};
     const pages = pagePaths.reduce((result, pagePath)=>{
@@ -185,6 +185,10 @@ function createPagesMapping({ isDev, pageExtensions, pagePaths, pagesType, pages
         if (pagePath.endsWith(".d.ts") && pageExtensions.includes("ts")) {
             return result;
         }
+        const ignorePagesArray = Array.isArray(ignorePages) ? ignorePages : ignorePages ? [ignorePages] : [];
+        if (ignorePagesArray.some((regex) => new RegExp(regex).test(pagePath))) {
+            return result;
+        }
         let pageKey = getPageFromPath(pagePath, pageExtensions);
         if (isAppRoute) {
             pageKey = pageKey.replace(/%5F/g, "_");
diff --git a/dist/build/index.js b/dist/build/index.js
index 353eaa712e453b4b8185cf8b9bf799a36194c6fe..04ce0b11247dc691711643c2a641fbbe65d6aa03 100644
--- a/dist/build/index.js
+++ b/dist/build/index.js
@@ -329,6 +329,7 @@ async function build(dir, reactProductionProfiling = false, debugOutput = false,
             const mappedPages = nextBuildSpan.traceChild("create-pages-mapping").traceFn(()=>(0, _entries.createPagesMapping)({
                     isDev: false,
                     pageExtensions: config.pageExtensions,
+                    ignorePages: config.ignorePages,
                     pagesType: "pages",
                     pagePaths: pagesPaths,
                     pagesDir
diff --git a/dist/server/config-schema.js b/dist/server/config-schema.js
index dae2dec14daae7feb32329109104a54574783a30..f21da9c215e33f78e0cdd4e0bb34c094f3d2962a 100644
--- a/dist/server/config-schema.js
+++ b/dist/server/config-schema.js
@@ -407,6 +407,10 @@ const configSchema = _zod.z.lazy(()=>_zod.z.strictObject({
             "export"
         ]).optional(),
         outputFileTracing: _zod.z.boolean().optional(),
+        ignorePages: _zod.z.union([
+            _zod.z.string(),
+            _zod.z.array(_zod.z.string()).min(1)
+        ]).optional(),
         pageExtensions: _zod.z.array(_zod.z.string()).min(1).optional(),
         poweredByHeader: _zod.z.boolean().optional(),
         productionBrowserSourceMaps: _zod.z.boolean().optional(),
diff --git a/dist/server/config-shared.d.ts b/dist/server/config-shared.d.ts
index 2d93f521495a5c24478714a19d8089487d106b9b..21051445d0937006572701ea70277909e590b175 100644
--- a/dist/server/config-shared.d.ts
+++ b/dist/server/config-shared.d.ts
@@ -488,6 +488,11 @@ export interface NextConfig extends Record<string, any> {
      *
      */
     outputFileTracing?: boolean;
+    /**
+     * Regex that define pages to ignore while building
+     *
+     */
+    ignorePages?: string | string[];
     /**
      * Timeout after waiting to generate static pages in seconds
      *
diff --git a/dist/server/config.js b/dist/server/config.js
index 541fa7863a2017619bce7d11a40de9cd7f017310..fbde5170cd49c391e2003732d17810a4d6eb2006 100644
--- a/dist/server/config.js
+++ b/dist/server/config.js
@@ -234,6 +234,21 @@ function assignDefaults(dir, userConfig, silent) {
                 }
             });
         }
+        if (key === "ignorePages") {
+            if (Array.isArray(value)) {
+                if (!value.length) {
+                    throw new Error(`Specified ignorePages is an empty array. Please update it with the relevant strings or remove it.`);
+                }
+                value.forEach((element)=>{
+                    if (typeof element !== 'string') {
+                        throw new Error(`Specified ignorePages is not an array of string, found "${element}" of type "${typeof element}". Please update this config or remove it.`);
+                    }
+                });
+            } else if (typeof value !== 'string') {
+                throw new Error(`Specified ignorePages is neither an array nor a string, found "${value}" of type "${typeof value}". Please update this config or remove it.`);
+            }
+            
+        }
         if (!!value && value.constructor === Object) {
             currentConfig[key] = {
                 ..._configshared.defaultConfig[key],

```

**Comment by JasonSimms** [comment]
Bump for visibility i still dont see this working with next.config.js exclude. (   "next": "14.2.5" )

**Comment by Virous77** [comment]
@JasonSimms if you are using typescript just add folder name in `tsconfig.json` - `"exclude": ["node_modules", "not-build"]
` like that.

**Reply by volkan0x** [reply]
doesn't work

**Reply by kristian-nst** [reply]
Why would i want to exclude files (while im coding?)
We just wanna exclude it when building.

**Reply by jaandrews** [reply]
Watching all the files can lead to infinite loops. I've run into issues with this when running a script in a nextjs project with npx in the terminal. Command looks something like when run in a visual studio code terminal
```javascript
npx run ./folder-to-ignore/updateTxt.js
```
After running the command, it logs
```
Watching C:\path\to\directory and all sub-directories not excluded by your .gitignore. Will not monitor dotfiles.
```
It then modifies, a txt file, but because that file is being watched, the whole command is restarted and run again. Usually, this wouldn't lead to an infinite loop, but this particular command calls an external api whose data changes each time it's called, so the file changes every time.

Tried adding the directory to the excludes array in tsconfig.json, but it is still watched. 

It is possible to avoid this issue by ignoring the files in gitignore, but these files that developers can also manually tweak, and so they should be in version control. The files generated are also only used with the nextjs project, so it doesn't really make sense to move them outside of the project to avoid the watch.

**Comment by cookieukw** [comment]
Hey everyone, I'm here to help anyone facing the same problem I had. But first, let me give you some context so you understand what was going on.

Basically, I had a Vite project integrated with Capacitor. There were some issues—not exactly bugs, but limitations. What I really wanted was to use Capacitor (which is part of the Ionic ecosystem) to build a mobile version.

At that time, I had two separate repositories. One was the mobile app built with Vite and integrated with Capacitor. The other was an API built in Next.js, which served as the backend.

My app is a book-related app, so it relied on an API—this API was built using Next.js version 13. Meanwhile, the mobile app was using Vite. Eventually, I decided to migrate everything from Vite to Next.js, which led to two different Next.js projects: one using version 14 and the other using version 13. They were completely separate—one for the API and one for the frontend/mobile app.

Then, earlier this year, I decided to merge both projects into one. I spent hours—actually days—working on the migration, tweaking configurations, researching, and testing. In the end, I managed to create a single project with both the frontend and the API. I was also able to build the mobile version without including the API files in the build.

However, at some point, I lost that working configuration(to build static files without errors on api folder). I’m not exactly sure what I changed. More recently, the Play Store started bothering me to update the SDK(to update app, wich used capacitor). I had updated a bunch of stuff on the site, and at some point, I lost the config. Somehow, everything got reset.

Now, every time I try to generate a build for the mobile version, Next.js throws an error about `generateStaticParams` or something like that. It seems like it's trying to include the API routes in the static build, which shouldn't happen(after the merged project)

After digging around for a while, I found this topic. It reminded me that I had already solved this same issue last month(when i try make  next + capacitor works) So I went back and investigated one of my other test apps, which also used Next.js + Capacitor (which requires static files to run properly). And in that project, I had used the following `next.config.ts` setup:

```ts
import type { NextConfig } from "next";

const nextConfig: NextConfig = {
  eslint: {
    ignoreDuringBuilds: true, // disable eslint checks on build
  },
  typescript: {
    ignoreBuildErrors: true, // ignore TypeScript errors on build
  },
  images: {
    unoptimized: true, // disable image optimization (which requires a server)
  },
  distDir: "build",
  output: "export", // export static files
};

module.exports = nextConfig;
```

Just a reminder—in my case, I updated the code to fetch the API directly from my website’s domain, which serves the API. I’m not using relative routes. For example:

```js
fetch("https://*.vercel.app/api/route")
```

instead of:

```js
fetch("/api/route")
```
